lib/node_modules/@stdlib/array/base/join/lib/main.js

```diff
@@ -0,0 +1,136 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+
+
+// FUNCTIONS //
+
+/**
+* Tests whether an object has a specified method.
+*
+* @private
+* @param {Object} obj - input object
+* @param {string} method - method name
+* @returns {boolean} boolean indicating whether an object has a specified method
+*
+* @example
+* var bool = hasMethod( [], 'join' );
+* // returns true
+*
+* @example
+* var bool = hasMethod( [], 'beep' );
+* // returns false
+*/
+function hasMethod( obj, method ) {
+	return ( typeof obj[ method ] === 'function' );
+}
+
+/**
+* Returns a string which is the concatenation of all the elements in an array with a given separator when input is an accesor array.
+*
+* @private
+* @param {Object} x - input array object
+* @param {integer} separator - separator
+* @returns {string} output array
+*
+* @example
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = arraylike2object( toAccessorArray( [ 1, 2, 3, 4 ] ) );
+*
+* var out = accessors( x, ',' );
+* // returns '1,2,3,4'
+*/
+function accessors( x, separator ) {
+	var output;
+	var data;
+	var get;
+	var i;
+	data = x.data;
+	get = x.accessors[ 0 ];
+	output = '';
+	for ( i = 0; i < data.length; i++ ) {
+		if ( typeof get( data, i ) === 'undefined' || get( data, i ) === null ) {
+			output += separator;
```

Comment:
What happens if `null` or `undefined` is the last element? Here you would append an additional separator, which would be incorrect.